micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java

```diff
@@ -91,44 +99,41 @@ public void bindTo(MeterRegistry registry) {
             return;
         }
 
-        double maxOldGen = getOldGen().map(mem -> getUsageValue(mem, MemoryUsage::getMax)).orElse(0.0);
+        double maxLongLivedPoolBytes = getLongLivedHeapPool().map(mem -> getUsageValue(mem, MemoryUsage::getMax)).orElse(0.0);
 
-        AtomicLong maxDataSize = new AtomicLong((long) maxOldGen);
+        AtomicLong maxDataSize = new AtomicLong((long) maxLongLivedPoolBytes);
         Gauge.builder("jvm.gc.max.data.size", maxDataSize, AtomicLong::get)
             .tags(tags)
-            .description("Max size of old generation memory pool")
+            .description("Max size of long-lived heap memory pool")
             .baseUnit(BaseUnits.BYTES)
             .register(registry);
 
         AtomicLong liveDataSize = new AtomicLong();
 
         Gauge.builder("jvm.gc.live.data.size", liveDataSize, AtomicLong::get)
             .tags(tags)
-            .description("Size of old generation memory pool after a full GC")
+            .description("Size of long-lived heap memory pool after reclamation")
             .baseUnit(BaseUnits.BYTES)
             .register(registry);
 
-        Counter promotedBytes = Counter.builder("jvm.gc.memory.promoted").tags(tags)
-            .baseUnit(BaseUnits.BYTES)
-            .description("Count of positive increases in the size of the old generation memory pool before GC to after GC")
-            .register(registry);
-
         Counter allocatedBytes = Counter.builder("jvm.gc.memory.allocated").tags(tags)
             .baseUnit(BaseUnits.BYTES)
-            .description("Incremented for an increase in the size of the young generation memory pool after one GC to before the next")
+            .description("Incremented for an increase in the size of the (young) heap memory pool after one GC to before the next")
             .register(registry);
 
+        Counter promotedBytes = (oldGenPoolName == null) ? null : Counter.builder("jvm.gc.memory.promoted").tags(tags)
```

Comment:
Promoted only makes sense with generational regions, so we only register the metric when using a collector with an old region.